docs/source/guides/area_detectors.rst

```diff
@@ -0,0 +1,610 @@
+
+**Fast CCD Pattern:**
+
+.. code-block:: python
+
+    # devices/fastccd.py - Fast CCD configuration
+    from ophyd.areadetector import DetectorBase
+    from ophyd.areadetector.cam import FastCCDDetectorCam
+    from ophyd.areadetector.plugins import HDF5Plugin_V34
+    from ophyd import Component as Cpt
+
+    class FastCCDDetector(DetectorBase):
+        """Fast CCD detector with HDF5 file writing."""
+
+        cam = Cpt(FastCCDDetectorCam, ":cam1:")
+        hdf5 = Cpt(HDF5Plugin_V34, ":HDF1:",
+                   write_path_template="/data/%Y/%m/%d/")
+
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+
+            # Fast CCD specific configuration
+            self.cam.fccd_fw_enable.put(1)  # Enable firmware
+            self.cam.fccd_sw_enable.put(1)  # Enable software
+
+**Area Detector with Custom Processing:**
+
+.. code-block:: python
+
+    # devices/processing_detector.py - Custom image processing
+    from ophyd.areadetector import DetectorBase
+    from ophyd.areadetector.plugins import ProcessPlugin_V34, ROIPlugin_V34
+    from ophyd import Component as Cpt
+
+    class ProcessingDetector(DetectorBase):
+        """Detector with real-time image processing."""
+
+        # Multiple ROIs for different sample regions
+        roi1 = Cpt(ROIPlugin_V34, ":ROI1:")
+        roi2 = Cpt(ROIPlugin_V34, ":ROI2:")
+        roi3 = Cpt(ROIPlugin_V34, ":ROI3:")
+
+        # Image processing
+        proc1 = Cpt(ProcessPlugin_V34, ":Proc1:")
+
+        def setup_rois(self, sample_positions):
+            """Configure ROIs for different sample positions."""
+            for i, (roi, pos) in enumerate(zip([self.roi1, self.roi2, self.roi3],
+                                               sample_positions)):
+                roi.min_x.put(pos['x'] - pos['width']//2)
+                roi.min_y.put(pos['y'] - pos['height']//2)
+                roi.size_x.put(pos['width'])
+                roi.size_y.put(pos['height'])
+
+Plugin Configuration Patterns
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+**File Writing Plugins:**
+
+.. code-block:: python
+
+    # devices/file_writers.py - Advanced file writing
+    from ophyd.areadetector.plugins import HDF5Plugin_V34, TIFFPlugin_V34
+    from ophyd import Component as Cpt
+    from pathlib import Path
+    import datetime
+
+    class MultiFormatDetector(DetectorBase):
+        """Detector that saves in multiple formats."""
+
+        hdf5 = Cpt(HDF5Plugin_V34, ":HDF1:")
+        tiff = Cpt(TIFFPlugin_V34, ":TIFF1:")
+
+        def configure_file_writing(self, experiment_name, sample_name):
+            """Configure file paths and names."""
+
+            # Create date-based directory structure
+            today = datetime.datetime.now()
+            data_path = Path(f"/data/{today.year:04d}/{today.month:02d}/{today.day:02d}")
+
+            # HDF5 for analysis
+            hdf5_path = data_path / "hdf5"
+            self.hdf5.file_path.put(str(hdf5_path))
```

Comment:
A peculiar detail about the EPICS area detector FilePath support conflicts at this point.  We must be certain to include a trailing path delimiter `/`.  Without this, the `.put()` will hang and eventually timeout.  Internally, the EPICS IOC will add this trailing path delimiter to the PV value.  A string representation of a pathlib object strips any trailing delimiters. 

Since ophyd waits for the PV value to reach the intended target value, we must supply a value that will match the EPICS result.

```suggestion
            self.hdf5.file_path.put(str(hdf5_path) + "/")
```